src/rgw/rgw_common.h

```diff
@@ -1584,7 +1584,12 @@ class rgw_obj {
       return;
     }
 
-    init_ns(b, key.name.substr(pos + 1), key.name.substr(1, pos -1));
+    // skip setting ns if object itself ends with an underscore
```

Comment:
@theanalyst it's not clear to me why we should even be here in that case. In the _temp_ example, we should have escaped the first underscore, so it should have turned into __temp_, and we should have returned earlier. Sounds to me that we fail to encode correctly.